bundles/org.openhab.binding.doorbird/README.md

```diff
@@ -0,0 +1,239 @@
+|-----------|-------------------|-------------|
+| ID | Required | The ID number of the Doorbird device. This is usually a single digit (e.g. 1). |
+| Hostname | Required | The ID number of the Doorbird device. This is usually a single digit (e.g. 1). |
+| User ID | Required | User Id of a Doorbird user that has permissions to access the camera, motion sensor, etc. The User ID and Password must be created using the Doorbird smart phone application.
+| Password | Required | Password of a Doorbird user. |
+| Image Refresh Rate | Optional | Rate at which image channel should be automatically updated. Leave field blank (default) to disable refresh. |
+| Doorbell Off Delay | Optional | Number of seconds to wait before setting doorbell channel OFF after a doorbell event. Leave field blank to disable. |
+| Motion Off Delay | Optional | Number of seconds to wait before setting motion channel OFF after a motion event. Leave field blank to disable. |
+| Montage Number of Images | Required | Number of images to include in the doorbell and motion montage images. Default is 0. |
+| Montage Scale Factor | Required | Percent scaling factor for montage image. Default is 100. |
+
+
+## Channels
+
+The following channels are supported by the binding.
+
+| Channel ID | Item Type | Description |
+|------------|-----------|-------------|
+| doorbell | Trigger | Generated PRESSED event when doorbell is pressed |
+| doorbellTimestamp | DateTime | Timestamp when doorbell was pressed |
+| doorbellImage | Image | Image captured when the doorbell was pressed |
+| motion | Trigger | Generated TRIGGERED event when motion is detected |
+| motionTimestamp | DateTime | Timestamp when motion sensor was triggered |
+| motionImage | Image | Image captured when motion was detected |
+| light | Switch | Activates the light relay |
+| openDoor1 | Switch | Activates the door 1 relay |
+| openDoor2 | Switch | Activates the door 2 relay (D210x only) |
+| getImage | Switch | Force an update to the Image channel |
+| image | Image | Image from the doorbird camera |
+| imageTimestamp | DateTime | Time when image was captured from device |
+| doorbellHistoryIndex | Number | Index of historical image for doorbell press |
+| doorbellHistoryTimestamp | DateTime | Time when doorbell was pressed for history image |
+| doorbellHistoryImage | Image | Historical image for doorbell press |
+| motionHistoryIndex | Number | Index of Historical image for motion |
+| motionHistoryTimestamp | DateTime | Time when motion was detected for history image |
+| motionHistoryImage | Image | Historical image for motion sensor |
+| doorbellMontage | Image | Concatenation of first n doorbell history images |
+| motionMontage | Image | Concatenation of first n motion history images |
+| sipHangup | Switch | Hangup SIP call |
+| restart | Switch | Restart the doorbird |
+
+## Profiles
+
+A Switch profile is available for the doorbell and motion channels that will cause ON/OFF 
```

Comment:
@cweitkamp Regarding your point #1 above, I'm not familiar with the `system.rawbutton` channel type nor the `rawbutton-toggle-switch` profile.

If I change the channel type of my `doorbell` channel to `system.rawbutton`, does that mean I can completely delete the profile package in my source tree? And, other than changing the channel type to `system.rawbutton`, is there anything else I need to do to use the `rawbutton-toggle-switch` profile?

Reply:
And, in the case of your point 2 above, we would 

- add `TRIGGERED` and `UNTRIGGERED` to [this file](https://github.com/openhab/openhab-core/blob/48d873a32ba0901a9b58f04c541f0f858fbfcd51/bundles/org.openhab.core.thing/src/main/java/org/eclipse/smarthome/core/thing/CommonTriggerEvents.java#L27), and

- add a new profile to [this file](https://github.com/openhab/openhab-core/blob/8d5d78937116741cc10313a47b057414f2708a15/bundles/org.openhab.core.thing/src/main/java/org/eclipse/smarthome/core/thing/profiles/SystemProfiles.java) (although I'm not exactly sure how this would be defined).

Anything else?

Reply:
Let's have a look into the avmfritz binding how it is working:

Definition:
https://github.com/openhab/openhab2-addons/blob/f39c7b46e1837b9682eb32ca58c78eebebe47629/bundles/org.openhab.binding.avmfritz/src/main/resources/ESH-INF/thing/thing-types.xml#L213-L216

The handler uses the `triggerChannel(channelUID, event)` method to trigger a `CommonTriggerEvents.PRESSED` or `CommonTriggerEvents.RELEASED` event. The `rawbutton-toggle-switch` profiles is listening to the first event and toggles a switch.

A typical item definition will look like this:
```
   channel="avmfritz:HAN_FUN_SWITCH:<BRIDGE_ID>:<THING_ID>:press"[profile="rawbutton-toggle-switch"]
```

That's it. I would add a note and link to the profile section in the [items](https://www.openhab.org/docs/configuration/items.html#profiles) page of the official docs.

> does that mean I can completely delete the profile package in my source tree? 
Yes,  you can remove the profile package from your binding.

Now I can see the difference. You need a profile which act a little bit different. `PRESSED` => `ON` and `RELEASED` => `OFF`. Not a toggle on `PRESSED`. Another potential system default profile.

> add TRIGGERED and UNTRIGGERED to this file
Correct. But as mentioned in https://github.com/openhab/openhab2-addons/pull/5857#discussion_r331658260 I'm not sure about the meaning of `UNTRIGGERED`.

> add a new profile to this file (although I'm not exactly sure how this would be defined).
Definition has to be done in the linked file. You can find the factory and available handlers in the [internal package](https://github.com/openhab/openhab-core/tree/8d5d78937116741cc10313a47b057414f2708a15/bundles/org.openhab.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/profiles).

Reply:
Ok, that all makes sense. I'll take a whack at the changes necessary to support this. I may have a few more questions as I get into it. I'm pretty comfortable working on bindings, but this is probably on the outer limit of my skill level.

Reply:
I've converted the `motion` channel back to a switch.

So, now w.r.t. the `doorbell` channel, I guess I need to create a new profile in core that will handle the `PRESSED` -> `ON` and `RELEASED` -> `OFF` switch functionality.

My only misgiving with this is that anyone who's not on a recent snapshot release won't be able to use the binding (and there are quite a few people using it), as the new profile won't be available.

Reply:
Yes, but a rule to imitate a similar behavior shouldn't be too complex as a replacement. Isn't it?

Reply:
Ok, @cweitkamp, I pushed an openhab-core PR to add the switch ON/OFF profile functionality. Please have a look and let me know if that's what you were expecting.

Reply:
Done